include/seqan3/argument_parser/detail/misc.hpp

```diff
@@ -0,0 +1,388 @@
+ *
+ */
+template <typename ...value_type>
+//!\cond
+requires requires (std::iostream s, value_type ...v) 
+{
+    (s << ... << v);
+}
+//!\endcond
+std::string _paste(value_type... values)
+{
+    std::stringstream stream;
+    (stream << ... << values);
+    return stream.str();
+}
+
+/*!\brief Returns the input type as a string (reflection).
+ * \tparam value_type The type to get as a std::string.
+ */
+template <typename value_type>
+inline std::string _get_type_name_as_string(value_type const & /**/)
+{
+    using type = std::decay_t<value_type>;
+    using types = meta::list<int8_t,
+                             uint8_t,
+                             int16_t,
+                             uint16_t,
+                             int32_t,
+                             uint32_t,
+                             int64_t,
+                             uint64_t,
+                             double,
+                             float,
+                             bool,
+                             char,
+                             std::string>;
+    std::vector<std::string> names{"INT (8 bit)",
+                                   "UNSIGNED (8 bit)",
+                                   "INT (16 bit)",
+                                   "UNSIGNED (16 bit)",
+                                   "INT (32 bit)",
+                                   "UNSIGNED (32 bit)",
+                                   "INT (64 bit)",
+                                   "UNSIGNED (64 bit)",
+                                   "DOUBLE",
+                                   "FLOAT",
+                                   "BOOL",
+                                   "CHAR",
+                                   "STRING"};
+
+    if constexpr (meta::in<types, type>::value)
+        return names[meta::find_index<types, type>::value];
+    else
+        return "UNKOWN_TYPE";
+}
+
+/*!\brief Returns the value_type of the input container as a string (reflection).
+ * \tparam container_type The container type for which to query it's value_type.
+ */
+template <typename container_type>
+requires container_concept<container_type> && !std::is_same_v<container_type, std::string>
+inline std::string _get_type_name_as_string(container_type const & /**/)
+{
+    typename container_type::value_type tmp;
+    return _get_type_name_as_string(tmp);
+}
+
+
+/*!\brief Formats the type of a value for the help page printing.
+ *
+ * \param value The value to deduct the type from.
+ */
+template <typename option_value_type>
+inline std::string _option_type_and_list_info(option_value_type const & value)
+{
+    return ("\\fI" + _get_type_name_as_string(value) + "\\fP");
+}
+
+/*!\brief Formats the container and its value_type for the help page printing.
+ *
+ * \param container The container to deduct the type from.
+ */
+template <typename container_type>
+requires container_concept<container_type> && !std::is_same_v<container_type, std::string>
```

Comment:
I actually asked Rene whether there is a "container" concept that does exclude std::string.
I'll take a look at input_range